### PR TITLE
Added support for getting optional value #33

### LINKED ISSUE #33: Add new method for getting optional value
Add new method `std::optional<T> getOptional<T>(const std::string& key)` for getting optional values.
If key does not exist or is null it should return std::nullopt, otherwise value with type T it is castable to that.

### PR DESCRIPTION
**Pull request by srivastava-yash**
Added `std::optional<T> getOptional<T>(const std::string& key)` for getting optional values.
If key does not exist or is null it should return std::nullopt, otherwise value with type T it is castable to that

### CODE REVIEW

File: src/Config.cpp [hunk 1]
```diff
@@ -56,6 +57,47 @@ T Config::get(const std::string& keyPath)
...
+    if (it == values.end())
+    {
+        return std::nullopt;
+    }
+
+    const auto& value = it->second;
+
+    if (value.index() == 0)
+    {
+        throw std::runtime_error("Config key " + keyPath + " has value of nullptr.");
```
Comment (cieslarmichal) [inline src/Config.cpp]: Return nullopt

**Review by cieslarmichal — CHANGES_REQUESTED** [review]
Good job, one little change

File: src/Config.cpp (hunk 1 above)
Comment (srivastava-yash) [reply src/Config.cpp]: resolved

**Review by cieslarmichal — APPROVED** [review]
Nice!

**Pull request merged**